sympy/physics/mechanics/kane.py

```diff
@@ -139,6 +147,16 @@ class KanesMethod(_Methods):
     done using `lambda A, b: tuple(linsolve((A, b)))[0]`, where we select the first
     solution as our system should have only one unique solution.
 
+    Care must be taken in the arrangement of the generalized coordinates when
+    linear and complex constraints are used: the generalized speeds on which the
+    complex constraints depend must be at the end of the list of the dependent
+    generalized speeds. This affects the arrangement of the generalized
+    coordinates. KanesMethod internally aranges the the generalized coordinates
+    in the following order:
+    y = q_independent + q_dependent +u_independent + u_dependent, and naturally
+    d/dt(q_independent + q_dependent) = u_independent + u_dependent)
+    must hold.
```

Comment:
Any way we can avoid this and do the right thing regardless of the order?

Reply:
> Any way we can avoid this and do the right thing regardless of the order?

I do not think so, or minimum I could to find a way. When Ars and/or Asstilde are formed they must be formed with u_dep on which they depend. I took me a while to find this problem.